firmware/command.cpp

```diff
@@ -1,156 +1,548 @@
-			Serial.flush();
-			return; // Early return, unidentified command
-	}
-	// Parse out parameters
-	while( **buf_ptr ) {
-		skip_whitespace( buf_ptr );
-		switch( **buf_ptr ) {
-			case 'X':
-				(*buf_ptr)++;
-				c->x = (get_float( buf_ptr ) * 100.0);
-				break;
-			case 'Y':
-				(*buf_ptr)++;
-				c->y = (get_float( buf_ptr ) * 100.0);
-				break;
-			case 'E':
-				(*buf_ptr)++;
-				c->e = get_int( buf_ptr );
-				break;
-			case 'F':
-				(*buf_ptr)++;
-				c->f = get_float( buf_ptr );
-				break;
-			case 'I':
-				(*buf_ptr)++;
-				c->i = (get_float( buf_ptr ) * 100.0);
-				break;
-			case 'J':
-				(*buf_ptr)++;
-				c->j = (get_float( buf_ptr ) * 100.0);
-				break;
-			default:
-				(*buf_ptr)++;
-				break;
-		}
-	}
-	return;
+#ifdef DEBUG_OUTPUT
+  Serial.println( F("// G command, reading numeric code..") );
+#endif
+  c->cmd_class = 'G';
+  // Parse out command ID
+  switch ( get_int( buf_ptr ) ) {
+    case 0:
+      c->cmd_id = 0;
+      break;
+    case 1:
+      c->cmd_id = 1;
+      break;
+    case 2:
+      c->cmd_id = 2;
+      break;
+    case 3:
+      c->cmd_id = 3;
+      break;
+    case 4:
+      c->cmd_id = 4; // G4: Dwell
+      break;
+    case 5: // Raster Right
+      c->cmd_id = 5;
+      break;
+    case 6: //Raster Left
+      c->cmd_id = 6;
+      break;
+    case 7: // Raster up
+      c->cmd_id = 7;
+      break;
+    case 8: // Raster down
+      c->cmd_id = 8;
+      break;
+    case 20:
+      c->cmd_id = 20;
+      break;
+    case 21:
+      c->cmd_id = 21;
+      break;
+    case 28:
+      c->cmd_id = 28;
+      return; // Early return, G28 ignores parameters
+    case 30:
+      c->cmd_id = 30;
+      return; // Early return, G30 ignores parameters
+    case 90:
+      c->cmd_id = 90;// G90 - Absolute positioning
+      break;
+    case 91:
+      c->cmd_id = 91; // G91 - Relative positioning
+      break;
+    case 92:
+      c->cmd_id = 92; // G92 - Set home Position
+      break;
+    case 93:
+      c->cmd_id = 93; // G93 - Set Table Position
+      break;
+
+    default:
+      Serial.print( F("// warning: unknown command " ) );
+      Serial.print(c->c) ; Serial.print(" ");
+      Serial.println( line );
+      Serial.flush();
+      return; // Early return, unidentified command
+  }
+  // Parse out parameters
+  parseOutParameters(c, buf_ptr);
+  return;
 }
 
+
+
+void parseOutParameters( Command * c, char **buf_ptr) {
+  // Parse out parameters
+  while ( **buf_ptr ) {
+    skip_whitespace( buf_ptr );
+    switch ( toupper(**buf_ptr) ) {
+      case 'X':
+        (*buf_ptr)++;
+        c->x =  (get_float( buf_ptr ) * 100.0);
+
+        break;
+      case 'Y':
+        (*buf_ptr)++;
+        c->y =  (get_float( buf_ptr ) * 100.0);
+
+        break;
+#ifdef Z_DRIVER_ENABLED
+      case 'Z':
+        (*buf_ptr)++;
+        c->z =  (get_float( buf_ptr ) * 100.0);
+
+        break;
+#endif
+#ifdef E_DRIVER_ENABLED
+      case 'E':
+        (*buf_ptr)++;
+        c->e = get_int( buf_ptr );
```

Comment:
Nice catch,  but the intention is to use the extruder as another axis, so it should use the same syntax as X,Y,Z